lib/helpers/convert$personaIdent.js

```diff
@@ -8,36 +10,46 @@ import { getConnection } from 'lib/connections/mongoose';
  * @param {any} identifier
  * @return {any}
  */
-const convertOpValue = (identifier) => {
-  const PersonaIdentifier = getConnection().model('PersonaIdentifier');
+const convertOpValue = async (identifier, {
+  organisation
+} = {}) => {
+  const personaService = getService();
 
-  const { key, value } = identifier;
-  const query = {
-    identifiers: {
-      $elemMatch: { key, value }
-    },
-    persona: { $exists: true }
+  const filter = {
+    key: identifier.key.replace('persona.import.', ''),
+    value: identifier.value
   };
 
-  return PersonaIdentifier.find(query).select({persona:1}).lean().exec()
-    .then((personaIdentifiers) => {
-    const ids = _.without(personaIdentifiers.map(
-        personaIdentifier => personaIdentifier.persona
-      ), undefined, null);
-    return { $in: ids };
-  });
+  // todo, check when actually exsits.
+  let attributes;
+  try {
+    // TODO: change to use non connection api.
+    const result = await personaService.getAttributes({
+      filter,
+      organisation,
+      limit: Number.MAX_SAFE_INTEGER,
+      maxTimeMS: MAX_TIME_MS,
+      maxScan: MAX_SCAN
+    });
+    attributes = map(result.edges, ({ node }) => node);
+  } catch (err) {
+    logger.error('convert$personaIdent', err);
+    return;
+  }
+
+  return { $in: map(attributes, ({ personaId }) => personaId) };
```

Comment:
We had issues here where a `null` value (wrongly placed) onto idents previously would match on all statements that also did not have a persona. Could we do some similar filtering to before to remove undefined/null values to ensure we've got a clean set of values passed out to our `$in` query